---
title: "Microbiota-based model improves the sensitivity for detecting colonic lesions"
csl: bmc.csl
output:
  word_document:
    reference_docx: manuscript_format.docx
  html_document: default
bibliography: references.bibtex
---





**Authors:** Nielson T. Baxter^1^, Mack T. Ruffin IV^2^, Mary A.M. Rogers^3^, and Patrick D. Schloss^1\*^


**Affiliations:**  
^1^Department of Microbiology and Immunology, University of Michigan, Ann Arbor, Michigan.  
^2^Department of Family Medicine, University of Michigan, Ann Arbor, Michigan.  
^3^Department of Internal Medicine, University of Michigan, Ann Arbor, Michigan.  
^\*^Corresponding author

**Email addresses:**  
NTB: ntbaxter@umich.edu  
MTR: mruffin@med.umich.edu  
MAMR: maryroge@med.umich.edu  
PDS: pschloss@umich.edu  


```{r startup, echo=F, message=F, warning=F}
deps = c("pROC","randomForest","AUCRF","knitr","rmarkdown","vegan","gtools");
for (dep in deps){
  if (dep %in% installed.packages()[,"Package"] == FALSE){
    install.packages(as.character(dep), quiet=TRUE);
  }
  library(dep, verbose=FALSE, character.only=TRUE)
}
tax <- read.delim('data/glne007.0.03.trim.tax', sep='\t', header=T, row.names=1)
meta <- read.delim('data/metadata.tsv', header=T, sep='\t')
shared <- read.delim('data/glne007.final.an.unique_list.0.03.subsample.0.03.filter.shared', header=T, sep='\t')

all_data <- merge(meta, shared, by.x='sample', by.y='Group')
all_data$lesion <- factor(NA, levels=c(0,1))
all_data$lesion[all_data$dx=='adenoma' | all_data$dx=='cancer'] <- 1
all_data$lesion[all_data$dx=='normal'] <- 0
```

```{r pval, echo=F}
pval <- function(p){
  if(p < 0.001){p <- 'p<0.001'}
  else{p <- sprintf('p=%.1g', p)}
  return(p)
}

```

```{r figure1_otusOnly, warning=F, results='hide', echo=F, fig.width=7, fig.height=7, cache=T}
### Cancer vs Normal

otu_canc_data <- all_data[meta$dx!='adenoma',c('lesion',colnames(all_data)[grep('Otu[0123456789]', colnames(all_data))])]

set.seed(62515)
otu_canc_model <- AUCRF(lesion ~ ., data=otu_canc_data, ranking='MDA', ntree=500, pdel=0.05)
canc_opt_model <- otu_canc_model$RFopt
otu_canc_probs <- predict(canc_opt_model, type='prob')
otu_canc_roc <- roc(otu_canc_data$lesion~otu_canc_probs[,2])

canc_fit_roc <- roc(all_data[all_data$dx!='adenoma','lesion'] ~ all_data[all_data$dx!='adenoma','fit_result'])

pdf(file='results/figures/Figure1.pdf', width=12, height=10)
layout(rbind(c(1,2),c(3,4)))
par(oma=c(1,6,1,1))


### Adenoma vs Normal
otu_ade_data <- all_data[meta$dx!='cancer',c('lesion',colnames(all_data)[grep('Otu[0123456789]', colnames(all_data))])]

set.seed(071315)
otu_ade_model <- AUCRF(lesion ~ ., data=otu_ade_data, ntree=500, pdel=0.05)

otu_ade_probs <- predict(otu_ade_model$RFopt, type='prob')
otu_ade_probs <- (otu_ade_probs-0.14)/0.86
otu_ade_roc <- roc(otu_ade_data$lesion~otu_ade_probs[,2])
ade_fit_roc <- roc(all_data[all_data$dx!='cancer','lesion'] ~ all_data[all_data$dx!='cancer','fit_result'])

par(mar=c(4,4,1,1))
plot(c(1,0),c(0,1), type='l', lty=3, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='', xlab='')
plot(otu_ade_roc, col='grey30', lwd=2, add=T, lty=1)
plot(ade_fit_roc, col='grey30', lwd=2, add=T, lty=2)
mtext(side=2, text="Sensitivity", line=2.5)
mtext(side=1, text="Specificity", line=2.5)
legend('bottomright', legend=c(sprintf('FIT - AUC: %.3g', ade_fit_roc$auc), sprintf('Microbiota - AUC: %.3g', otu_ade_roc$auc), pval(roc.test(ade_fit_roc,otu_ade_roc)$p.value)), bty='n', xjust=0, lty=c(2,1,NA), col='grey30')

otu_ade_spec <- coords(ade_fit_roc, x=100, input='thr', ret='spec')
otu_ade_sens <- coords(otu_ade_roc, x=otu_ade_spec, input='spec', ret='sens')
otu_ade_thr <- coords(otu_ade_roc, x=otu_ade_spec, input='spec', ret='thr')

points(otu_ade_spec,otu_ade_sens, pch=16, cex=1.5)
points(coords(ade_fit_roc, x=100, input='thr', ret='spec'),coords(ade_fit_roc, x=100, input='thr', ret='sens'), pch=10, cex=1.5)
mtext('A', at=1.15, font=2)

ade_res <- data.frame(fit=jitter(all_data[all_data$dx=='adenoma','fit_result']+0.1, amount=0.03),micro=otu_ade_probs[otu_ade_data$lesion==1,2])

plot(ade_res, log='x', ylab='', xlab='', xaxt='n', ylim=c(0,1), pch=21, bg='grey95', col='black')
points(ade_res[ade_res$fit>100 | ade_res$micro>otu_ade_thr,], pch=21, bg='grey60', col='black')
points(ade_res[ade_res$fit>100 & ade_res$micro>otu_ade_thr,], pch=21, bg='black', col='black')
axis(1, labels=c(0,1,10,100,1000), at=c(0.1,1.1,10,100,1000))
mtext(side=2, text='Microbiota Model - Probability of Adenoma', line=2.5)
mtext(side=1, text='FIT Result (ng/ml)', line=2.5)
abline(h=otu_ade_thr, lty=3)
abline(v=100, lty=3)
legend('bottomleft', legend=c('Detected by both tests', 'Detected by one test', 'Undetected', 'Test cutoffs'), pch=c(21,21,21,NA), lty=c(NA,NA,NA,3), pt.bg=c('black','grey60','grey90',NA), cex=0.8)
mtext('B', at=0.04, font=2, side=3)

ade_pos <- cbind(all_data[all_data$dx=='adenoma','fit_result']>100, otu_ade_probs[otu_ade_data$lesion==1,2]>otu_ade_thr)
colnames(ade_pos) <- c('FIT','Microbiota')
ade_table <- 100*table(as.data.frame(ade_pos))/sum(all_data$dx=='adenoma')
text(x=c(1,1,500,500), y=c(0.9,0.3,0.9,0.3), labels=sprintf('%.3g%%',ade_table[c(3,1,4,2)]))


par(mar=c(4,4,1,1))
plot(c(1,0),c(0,1), type='l', lty=3, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='', xlab='')
plot(otu_canc_roc, col='grey30', lwd=2, add=T)
plot(canc_fit_roc, col='grey30', lwd=2, lty=2, add=T)
mtext(side=2, text="Sensitivity", line=2.5)
mtext(side=1, text="Specificity", line=2.5)
legend('bottomright', legend=c(sprintf('FIT - AUC: %.3g', canc_fit_roc$auc), sprintf('Microbiota - AUC: %.3g', otu_canc_roc$auc), pval(roc.test(canc_fit_roc,otu_canc_roc)$p.value)), bty='n', xjust=0, lty=c(2,1,NA), col='grey30')


otu_canc_sens <- coords(otu_canc_roc, x=coords(canc_fit_roc, x=100, input='thr', ret='spec'), input='spec', ret='sens')
otu_canc_spec <- coords(otu_canc_roc, x=coords(canc_fit_roc, x=100, input='thr', ret='spec'), input='spec', ret='spec')
otu_canc_thr <- coords(otu_canc_roc, x=coords(canc_fit_roc, x=100, input='thr', ret='spec'), input='spec', ret='thr')

points(otu_canc_spec,otu_canc_sens, pch=16, cex=1.5)
points(coords(canc_fit_roc, x=100, input='thr', ret='spec'),coords(canc_fit_roc, x=100, input='thr', ret='sens'), pch=10, cex=1.5)
mtext('C', at=1.15, font=2)

canc_res <- data.frame(fit=jitter(all_data[all_data$dx=='cancer','fit_result']+0.1, amount=0.01),micro=otu_canc_probs[otu_canc_data$lesion==1,2])

plot(canc_res, log='x', ylab='', xlab='', xaxt='n', ylim=c(0,1), pch=21, bg='grey95', col='black')
points(canc_res[canc_res$fit>100 | canc_res$micro>otu_canc_thr,], pch=21, bg='grey60', col='black')
points(canc_res[canc_res$fit>100 & canc_res$micro>otu_canc_thr,], pch=21, bg='black', col='black')
axis(1, labels=c(0,1,10,100,1000), at=c(0.1,1.1,10,100,1000))
mtext(side=2, text='Microbiota Model - Probability of Cancer', line=2.5)
mtext(side=1, text='FIT Result (ng/ml)', line=2.5)
abline(h=otu_canc_thr, lty=3)
abline(v=100, lty=3)
legend('bottomleft', legend=c('Detected by both tests', 'Detected by one test', 'Undetected', 'Test cutoffs'), pch=c(21,21,21,NA), lty=c(NA,NA,NA,3), pt.bg=c('black','grey60','grey90',NA), cex=0.8)
mtext('D', at=0.04, font=2, side=3)

canc_pos <- cbind(all_data[all_data$dx=='cancer','fit_result']>100, otu_canc_probs[otu_canc_data$lesion==1,2]>otu_canc_thr)
colnames(canc_pos) <- c('FIT','Microbiota')
canc_table <- 100*table(as.data.frame(canc_pos))/sum(all_data$dx=='cancer')
text(x=c(1,1,400,200), y=c(0.87,0.27,0.87,0.27), labels=sprintf('%.3g%%',canc_table[c(3,1,4,2)]))


mtext('Adenoma\nvs.\nNormal', las=1, side=2, at=0.8,outer=T, adj=0.5, line=3, font=2)
mtext('Cancer\nvs.\nNormal', las=1, side=2, at=0.3,outer=T, adj=0.5, line=3, font=2)
dev.off()
```

```{r figS1_adeModel, warning=F, echo=F, fig.width=7.5, fig.height=10, results='hide'}
pdf('results/figures/figS1.pdf', height=6, width=8)
layout(matrix(c(1,3,2,2), nrow=2))


#AUCRF curve
par(mar=c(5,5,1.5,1))
plot(otu_ade_model)
mtext('A', at=-90, font=2, side=3)

#Importance Plot
ade_otus <- otu_ade_model$Xopt
dotchart(rev(otu_ade_model$ranking[ade_otus]), labels=rev(tax[ade_otus, 'Label']), xlab='Mean Decrease Accuracy')
mtext('B', at=-0.014, font=2, side=3)

#Abundance stripchart or most predictive otus
ade_otus <- otu_ade_model$Xopt
ade_otus <- rev(ade_otus[1:5])
norm_abunds <- all_data[all_data$dx=='normal', ade_otus]/10000 + 1e-4
ade_abunds <- all_data[all_data$dx=='adenoma', ade_otus]/10000 + 1e-4

par(mar=c(4, 9, 1, 1))
plot(1, type="n", ylim=c(0,length(ade_otus)*2), xlim=c(1e-4,3), log="x", ylab="", xlab="Relative Abundance (%)", xaxt="n", yaxt="n")
index <- 1
for(i in ade_otus){
stripchart(at=index-0.35, jitter(norm_abunds[,i], amount=1e-5), pch=21, bg="royalblue1", method="jitter", jitter=0.2, add=T, cex=1, lwd=0.5)
  stripchart(at=index+0.35, jitter(ade_abunds[,i], amount=1e-5), pch=21, bg="orange", method="jitter", jitter=0.2, add=T, cex=1, lwd=0.5)
  segments(mean(norm_abunds[,i]),index-0.7,mean(norm_abunds[,i]),index, lwd=3)
  segments(mean(ade_abunds[,i]),index+0.7,mean(ade_abunds[,i]),index, lwd=3)
  index <- index + 2
}
axis(2, at=seq(1,index-2,2), labels=tax[ade_otus, 'Label'], las=1, line=-0.5, tick=F, cex.axis=0.8)
axis(1, at=c(1e-4, 1e-3, 1e-2, 1e-1, 1), label=c("0", "0.1", "1", "10", "100"))
legend('topright', legend=c("Adenoma", "Normal"), pch=c(21, 21), pt.bg=c("orange","royalblue1"), cex=0.7)
mtext('C', at=1e-7, font=2, side=3)


dev.off()
```

```{r figS2_otu_CVs, echo=F, cache=T, results='hide'}
##OTU adenoma model
cv_ade_data <- otu_ade_data[,c('lesion',otu_ade_model$Xopt)]

### Leave-1-out
  lv1_probs <- rep(NA,nrow(cv_ade_data))
  for(i in 1:nrow(cv_ade_data)){
   train <- cv_ade_data[-i,]
   test <- cv_ade_data[i,]
   temp_model <- AUCRF(lesion~., data=train, pdel=0.99, ntree=500)
   lv1_probs[i] <- predict(temp_model$RFopt, test, type='prob')[,2]
  }
  lv1_roc <- roc(cv_ade_data$lesion~lv1_probs)

### 10-fold CV
start <- proc.time()[3]
iters <- 100
cv10f_aucs <- c()
cv10f_all_resp <- c()
cv10f_all_pred <- c()
for(j in 1:iters){
  sampling <- sample(1:370,370,replace=F)
  cv10f_probs <- rep(NA,nrow(cv_ade_data))
  for(i in seq(1,370,37)){
    train <- cv_ade_data[sampling[-(i:(i+36))],]
    test <- cv_ade_data[sampling[i:(i+36)],]
    temp_model <- AUCRF(lesion~., data=train, pdel=0.99, ntree=500)
    cv10f_probs[sampling[i:(i+36)]] <- predict(temp_model$RFopt, test, type='prob')[,2]
  }
  cv10f_roc <- roc(cv_ade_data$lesion~cv10f_probs)
  cv10f_all_pred <- c(cv10f_all_pred, cv10f_probs)
  cv10f_all_resp <- c(cv10f_all_resp, cv_ade_data$lesion)
  cv10f_aucs[j] <- cv10f_roc$auc
}
cv10f_roc <- roc(cv10f_all_resp~cv10f_all_pred)
end <- proc.time()[3]
time <- end-start

pdf('results/figures/figS2.pdf', height=5.5, width=11)
layout(matrix(c(1,2),nrow=1))
par(mar=c(4.5,4.5,1,1))
plot(c(1,0),c(0,1), type='l', lty=3, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='Sensitivity', xlab='Specificity')
plot(otu_ade_roc, col='purple', lwd=2, add=T)
plot(lv1_roc, col='blue', lwd=2, add=T)
plot(cv10f_roc, col='green3', lwd=2, add=T)
mtext('A', at=1.12, font=2, side=3)
legend('bottomright', legend=c(sprintf('Out-of-bag (AUC: %.3g)',otu_ade_roc$auc),
                               sprintf('Leave-1-out (AUC: %.3g)',lv1_roc$auc),
                               sprintf('10-fold CV (AUC: %.3g)',cv10f_roc$auc),
                               sprintf('OOB vs Leave-1-out: p=%.2g', roc.test(otu_ade_roc,lv1_roc)$p.value),
                               sprintf('OOB vs 10-fold CV: p=%.2g', roc.test(otu_ade_roc,cv10f_roc)$p.value)
                               ),lty=c(1,1,1,0,0), lwd=2, col=c('purple','blue','green3'), bty='n')


## OTU Cancer model
cv_canc_data <- otu_canc_data[1:290,c('lesion',otu_canc_model$Xopt)]

### Leave-1-out
  lv1_probs <- rep(NA,nrow(cv_canc_data))
  for(i in 1:nrow(cv_canc_data)){
   train <- cv_canc_data[-i,]
   test <- cv_canc_data[i,]
   temp_model <- AUCRF(lesion~., data=train, pdel=0.99, ntree=500)
   lv1_probs[i] <- predict(temp_model$RFopt, test, type='prob')[,2]
  }
  lv1_roc <- roc(cv_canc_data$lesion~lv1_probs)

### 10-fold CV
start <- proc.time()[3]
iters <- 100
cv10f_aucs <- c()
cv10f_all_resp <- c()
cv10f_all_pred <- c()
for(j in 1:iters){
  sampling <- sample(1:290,290,replace=F)
  cv10f_probs <- rep(NA,nrow(cv_canc_data))
  for(i in seq(1,290,29)){
    train <- cv_canc_data[sampling[-(i:(i+28))],]
    test <- cv_canc_data[sampling[i:(i+28)],]
    temp_model <- AUCRF(lesion~., data=train, pdel=0.99, ntree=500)
    cv10f_probs[sampling[i:(i+28)]] <- predict(temp_model$RFopt, test, type='prob')[,2]
  }
  cv10f_roc <- roc(cv_canc_data$lesion~cv10f_probs)
  cv10f_all_pred <- c(cv10f_all_pred, cv10f_probs)
  cv10f_all_resp <- c(cv10f_all_resp, cv_canc_data$lesion)
  cv10f_aucs[j] <- cv10f_roc$auc
}
cv10f_roc <- roc(cv10f_all_resp~cv10f_all_pred)
end <- proc.time()[3]
time <- end-start

plot(c(1,0),c(0,1), type='l', lty=3, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='Sensitivity', xlab='Specificity')
plot(otu_canc_roc, col='purple', lwd=2, add=T)
plot(lv1_roc, col='blue', lwd=2, add=T)
plot(cv10f_roc, col='green3', lwd=2, add=T)
mtext('B', at=1.12, font=2, side=3)
legend('bottomright', legend=c(sprintf('Out-of-bag (AUC: %.3g)',otu_canc_roc$auc),
                               sprintf('Leave-1-out (AUC: %.3g)',lv1_roc$auc),
                               sprintf('10-fold CV (AUC: %.3g)',cv10f_roc$auc),
                               sprintf('OOB vs Leave-1-out: p=%.2g', roc.test(otu_canc_roc,lv1_roc)$p.value),
                               sprintf('OOB vs 10-fold CV: p=%.2g', roc.test(otu_canc_roc,cv10f_roc)$p.value)
                               ),lty=c(1,1,1,0,0), lwd=2, col=c('purple','blue','green3'), bty='n')
dev.off()
```

```{r figS3_cancModel, warning=F, echo=F, fig.width=7.5, fig.height=10, results='hide'}
pdf('results/figures/figS3.pdf', height=6, width=8)
layout(matrix(c(1,3,2,2), nrow=2))

#AUCRF curve
par(mar=c(5,5,1.5,1))
plot(otu_canc_model)
mtext('A', at=-90, font=2, side=3)

#Importance Plot
par(mar=c(4,0,1,1))
canc_otus <- otu_canc_model$Xopt
dotchart(rev(otu_canc_model$ranking[canc_otus]), labels=rev(tax[canc_otus, 'Label']), xlab='Mean Decrease Accuracy')
mtext('B', at=-0.014, font=2, side=3)

#Abundance stripchart or most predictive otus
canc_otus <- otu_canc_model$Xopt
canc_otus <- rev(canc_otus[1:7])
norm_abunds <- all_data[all_data$dx=='normal', canc_otus]/10000 + 1e-4
canc_abunds <- all_data[all_data$dx=='cancer', canc_otus]/10000 + 1e-4

par(mar=c(4, 10, 1, 1))
plot(1, type="n", ylim=c(0,length(canc_otus)*2), xlim=c(1e-4,1), log="x", ylab="", xlab="Relative Abundance (%)", xaxt="n", yaxt="n")
index <- 1
for(i in canc_otus){
stripchart(at=index-0.35, jitter(norm_abunds[,i], amount=1e-5), pch=21, bg="royalblue1", method="jitter", jitter=0.2, add=T, cex=1, lwd=0.5)
  stripchart(at=index+0.35, jitter(canc_abunds[,i], amount=1e-5), pch=21, bg="red", method="jitter", jitter=0.2, add=T, cex=1, lwd=0.5)
  segments(median(norm_abunds[,i]),index-0.7,median(norm_abunds[,i]),index, lwd=3)
  segments(median(canc_abunds[,i]),index+0.7,median(canc_abunds[,i]),index, lwd=3)
  index <- index + 2
}
axis(2, at=seq(1,index-2,2), labels=tax[canc_otus, 'Label'], las=1, line=-0.5, tick=F, cex.axis=0.8)
axis(1, at=c(1e-4, 1e-3, 1e-2, 1e-1, 1), label=c("0", "0.1", "1", "10", "100"))
legend('bottomright', legend=c("Cancer", "Normal"), pch=c(21, 21), pt.bg=c("red","royalblue1"), cex=0.7)
mtext('C', at=1e-7, font=2, side=3)

dev.off()
```

```{r figure2_MMT, warning=F, results='hide', echo=F, fig.width=7, fig.height=7, cache=T}
pdf('results/figures/figure2.pdf', width=10, height=5)

layout(matrix(c(1,1,2,3), nrow=1))
par(oma=c(1,1,2,1))

les_data <- all_data[,c('lesion','fit_result',colnames(all_data)[grep('Otu[0123456789]', colnames(all_data))])]

#generate lesion vs normal model
set.seed(53015)
les_model <- AUCRF(lesion ~ ., data=les_data, ntree=500, pdel=0.05, ranking='MDG')
opt_model <- les_model$RFopt
all_probs <- predict(opt_model, type='prob')
all_probs <- (all_probs-0.2)/0.8 #adjustment for wider spread of probabilities
les_roc <- roc(les_data$lesion~all_probs[,2])
les_fit_roc <- roc(les_data$lesion~les_data$fit_result)

#test model on cancer
canc_data <- les_data[all_data$dx!='adenoma',]
canc_probs <- all_probs[all_data$dx!='adenoma',]
canc_roc <- roc(canc_data$lesion ~ canc_probs[,2])
canc_fit_roc <- roc(canc_data$lesion ~ canc_data$fit_result)

ade_data <- les_data[all_data$dx!='cancer',]
ade_probs <- all_probs[all_data$dx!='cancer',]
ade_roc <-  roc(ade_data$lesion ~ ade_probs[,2])
ade_fit_roc <- roc(ade_data$lesion ~ ade_data$fit_result)

par(mar=c(4.5,4.5,1,1))
plot(c(1,0),c(0,1), type='l', lty=3, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='', xlab='', cex.axis=1.2)
plot(les_roc, col='darkred', lwd=2, add=T)
plot(canc_roc, col='red', lwd=2, add=T)
plot(ade_roc, col='orange', lwd=2, add=T)
plot(ade_fit_roc, col='orange', lwd=2, add=T, lty=2)
plot(canc_fit_roc, col='red', lwd=2, add=T, lty=2)
plot(les_fit_roc, col='darkred', lwd=2, add=T, lty=2)
mtext(side=2, text="Sensitivity", line=2.5, cex=1)
mtext(side=1, text="Specificity", line=2.5, cex=1)
mtext('A', at=1.12, font=2, side=3)
legend('bottomright', legend=c(sprintf('MMT-Les. vs Norm.: %.3g',les_roc$auc),
                               sprintf('FIT-Les. vs Norm.: %.3g',les_fit_roc$auc),
                               sprintf('MMT-Canc. vs Norm: %.3g',canc_roc$auc),
                               sprintf('FIT-Canc. vs Norm.: %.3g',canc_fit_roc$auc),
                               sprintf('MMT-Ade. vs Norm.: %.3g',ade_roc$auc),
                               sprintf('FIT-Ade. vs Norm.: %.3g',ade_fit_roc$auc)),lty=c(1,2,1,2,1,2), lwd=2, col=c('darkred','darkred','red','red','orange','orange'), bty='n', cex=0.8, seg.len=3)

points(coords(canc_fit_roc, x=100, input='threshold', ret='specificity'),coords(canc_fit_roc, x=100, input='threshold', ret='sensitivity'), col='red', cex=1.5, pch=10)
points(coords(les_fit_roc, x=100, input='threshold', ret='specificity'),coords(les_fit_roc, x=100, input='threshold', ret='sensitivity'), col='darkred', cex=1.5, pch=10)
points(coords(ade_fit_roc, x=100, input='threshold', ret='specificity'),coords(ade_fit_roc, x=100, input='threshold', ret='sensitivity'), col='orange', cex=1.5, pch=10)

cutoff <- coords(roc=canc_roc, x='best', best.method='y', ret=c('threshold'))
cutoff <- 0.57

points(coords(canc_roc, x=cutoff, input='threshold', ret='specificity'),coords(canc_roc, x=cutoff, input='threshold', ret='sensitivity'), col='red', cex=1.5, pch=16)
points(coords(les_roc, x=cutoff, input='threshold', ret='specificity'),coords(les_roc, x=cutoff, input='threshold', ret='sensitivity'), col='darkred', cex=1.5, pch=16)
points(coords(ade_roc, x=cutoff, input='threshold', ret='specificity'),coords(ade_roc, x=cutoff, input='threshold', ret='sensitivity'), col='orange', cex=1.5, pch=16)

par(mar=c(3, 3.5, 1, 1))
plot(1, type="n", xlim=c(0.5,3.5), ylim=c(0.07,5000), log="y", ylab="", xlab="", xaxt="n", yaxt="n")
stripchart(at=1, jitter(all_data[all_data$dx=='normal','fit_result']+.1, amount=0.05), vertical=T, method='jitter', jitter=0.4, pch=21, bg='royalblue1', add=T, cex=1.5)
stripchart(at=2, jitter(all_data[all_data$dx=='adenoma','fit_result']+.1, amount=0.05), vertical=T, method='jitter', jitter=0.4, pch=21, bg='orange', add=T, cex=1.5)
stripchart(at=3, jitter(all_data[all_data$dx=='cancer','fit_result']+.1, amount=0.05), vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T, cex=1.5)
axis(1, at=c(0.8,2,3.2), labels=c('Normal','Adenoma','Cancer'), cex.axis=1.2)
axis(2, at=c(0.1,1.1,10.1,100.1,1000.1), labels=c(0,1,10,100,1000), cex.axis=1.2)
mtext(side=2, text="FIT Result (ng/ml)", line=2.2, cex=0.8)
mtext('B', at=0, font=2, side=3)
abline(h=100, lty=2)

plot(1, type="n", xlim=c(0.5,3.5), ylim=c(0,1), ylab="", xlab="", xaxt="n", cex.axis=1.2)
stripchart(at=1, all_probs[all_data$dx=='normal',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='royalblue1', add=T, cex=1.5)
stripchart(at=2, all_probs[all_data$dx=='adenoma',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='orange', add=T, cex=1.5)
stripchart(at=3, all_probs[all_data$dx=='cancer',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T, cex=1.5)
mtext(side=2, text="MMT Probability of Lesion", line=2.2, cex=0.8)
mtext('C', at=0, font=2, side=3)
axis(1, at=c(0.8,2,3.2), labels=c('Normal','Adenoma','Cancer'), cex.axis=1.2)
abline(h=cutoff, lty=2)

dev.off()
```

```{r figS4_mmtOTUs, warning=F, echo=F, fig.width=7.5, fig.height=10, results='hide'}

les_otus <- les_model$Xopt
les_otus <- les_otus[les_otus!='fit_result']
les_otus <- rev(les_otus)

tax <- read.delim('data/glne007.0.03.trim.tax', sep='\t', header=T, row.names=1)
les_tax <- tax[les_otus, 'Label']

norm_abunds <- all_data[all_data$dx=='normal', les_otus]/10000 + 1e-4
les_abunds <- all_data[all_data$dx!='normal', les_otus]/10000 + 1e-4

pdf('results/figures/figS4.pdf', width=10, height=10)
layout(matrix(1:2, nrow=1), widths=c(1,2.5))

#Importance Plot
par(mar=c(4,1,1,1))
plot(1, type='n', xlim=c(3,0), ylim=c(0.5,length(les_otus)*1.18), ylab='', yaxt='n', xlab='')
barplot(height=les_model$ranking[les_otus], names.arg='', xlab='Variable Importance\n(Mean Decrease Gini)', horiz=T, add=T)
      
#Stripchart of abundances
par(mar=c(4,11,1,1))
plot(1, type="n", ylim=c(0.5,length(les_otus)*2-1), xlim=c(1e-4,1), log="x", ylab="", xlab="Relative Abundance (%)", xaxt="n", yaxt="n")
index <- 1
for(i in les_otus){
  stripchart(at=index-0.35, jitter(norm_abunds[,i], amount=1e-5), pch=21, bg="royalblue1", method="jitter", jitter=0.2, add=T, cex=0.8, lwd=0.5)
  stripchart(at=index+0.35, jitter(les_abunds[,i], amount=1e-5), pch=21, bg="red3", method="jitter", jitter=0.2, add=T, cex=0.8, lwd=0.5)
  segments(median(norm_abunds[,i]),index-0.75,median(norm_abunds[,i]),index+0.05, lwd=3)
  segments(median(les_abunds[,i]),index+0.75,median(les_abunds[,i]),index-0.05, lwd=3)
  index <- index + 2
}
axis(1, at=c(1e-4, 1e-3, 1e-2, 1e-1, 1), label=c("0", "0.1", "1", "10", "100"))
legend(x=.2,y=9, legend=c("Lesion", "Normal"), pch=c(21, 21), pt.bg=c("red3","royalblue1"))
mtext(les_tax, at=1:length(les_otus)*2-1, side=2, las=1, adj=0.5, line=6)
dev.off()
```

```{r figS5_mmtCV, echo=F, cache=T, results='hide'}
mmt_data <- les_data[,c('lesion',les_model$Xopt)]

### Leave-1-out
  lv1_probs <- rep(NA,nrow(mmt_data))
  for(i in 1:nrow(mmt_data)){
   train <- mmt_data[-i,]
   test <- mmt_data[i,]
   temp_model <- AUCRF(lesion~., data=train, pdel=0.99, ntree=500)
   lv1_probs[i] <- predict(temp_model$RFopt, test, type='prob')[,2]
  }
  lv1_roc <- roc(mmt_data$lesion~lv1_probs)

### 10-fold CV
start <- proc.time()[3]
iters <- 100
cv10f_aucs <- c()
cv10f_all_resp <- c()
cv10f_all_pred <- c()
for(j in 1:iters){
  sampling <- sample(1:490,490,replace=F)
  cv10f_probs <- rep(NA,nrow(mmt_data))
  for(i in seq(1,490,49)){
    train <- mmt_data[sampling[-(i:(i+48))],]
    test <- mmt_data[sampling[i:(i+48)],]
    temp_model <- AUCRF(lesion~., data=train, pdel=0.99, ntree=500)
    cv10f_probs[sampling[i:(i+48)]] <- predict(temp_model$RFopt, test, type='prob')[,2]
  }
  cv10f_roc <- roc(mmt_data$lesion~cv10f_probs)
  cv10f_all_pred <- c(cv10f_all_pred, cv10f_probs)
  cv10f_all_resp <- c(cv10f_all_resp, mmt_data$lesion)
  cv10f_aucs[j] <- cv10f_roc$auc
}
cv10f_roc <- roc(cv10f_all_resp~cv10f_all_pred)
end <- proc.time()[3]
time <- end-start

pdf('results/figures/figS5.pdf', height=5, width=6)
par(mar=c(4.5,4.5,1,1))
plot(c(1,0),c(0,1), type='l', lty=3, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='Sensitivity', xlab='Specificity')
plot(les_roc, col='purple', lwd=2, add=T)
plot(lv1_roc, col='blue', lwd=2, add=T)
plot(cv10f_roc, col='green3', lwd=2, add=T)

legend('bottomright', legend=c(sprintf('MMT out-of-bag (AUC: %.3g)',les_roc$auc),
                               sprintf('Leave-1-out (AUC: %.3g)',lv1_roc$auc),
                               sprintf('10-fold CV (AUC: %.3g)',cv10f_roc$auc),
                               sprintf('OOB vs Leave-1-out: p=%.2g', roc.test(les_roc,lv1_roc)$p.value),
                               sprintf('OOB vs 10-fold CV: p=%.2g', roc.test(les_roc,cv10f_roc)$p.value)
                               ),lty=c(1,1,1,0,0), lwd=2, col=c('purple','blue','green3'), bty='n')
dev.off()
```

```{r figure3_negativeFIT, echo=F, fig.width=8, fig.height=6, results='hide'}
### FIT vs MMT scatter plot

pdf('results/figures/Figure3.pdf', width=10, height=6)
layout(matrix(c(1,2), nrow=1), widths=c(3,2))
par(mar=c(4, 4, 2, 0.5))
plot(1, type="n", xlim=c(0.1,10000), ylim=c(0,1), log='x',ylab="", xlab="", xaxt='n')

palette(c('orange','red','royalblue1'))
points(all_data[,'fit_result'], all_probs[,2], pch=21, bg=all_data$dx)
points(jitter(rep(0.15,length(all_probs[all_data$fit_result==0,2])), factor=20), all_probs[all_data$fit_result==0,2], pch=21, bg=all_data[all_data$fit_result==0, 'dx'])
mtext(side=2, text="MMT Probability of Lesion", line=2.2, cex=1)
mtext(side=1, text="FIT (ng/ml)", line=2.2, cex=1)
axis(1, at=c(0.15,1.1,10.1,100.1,1000.1), labels=c(0,1,10,100,1000))
abline(h=cutoff, lty=2)
abline(v=100.1, lty=2)
legend('bottomright', legend=c('Normal','Adenoma','Cancer'), pt.bg=c('royalblue1','orange','red'), pch=21, bty='n', pt.cex=1)

mtext('A', at=0.03, font=2, side=3)

###Stripchart by FIT result
plot(1, type="n", xlim=c(0.5,7), ylim=c(0,1), ylab="", xlab="", xaxt="n")

stripchart(at=1, all_probs[all_data$dx=='normal' & all_data$fit_result >= 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='royalblue1', add=T)
stripchart(at=2, all_probs[all_data$dx=='adenoma' & all_data$fit_result >= 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='orange', add=T)
stripchart(at=3, all_probs[all_data$dx=='cancer' & all_data$fit_result >= 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)

stripchart(at=4.5, all_probs[all_data$dx=='normal' & all_data$fit_result < 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='royalblue1', add=T)
stripchart(at=5.5, all_probs[all_data$dx=='adenoma' & all_data$fit_result < 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='orange', add=T)
stripchart(at=6.6, all_probs[all_data$dx=='cancer' & all_data$fit_result < 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)

axis(1, at=c(2,5), labels=c('Positive FIT','Negative FIT'), cex=0.8)
mtext(side=2, text="MMT Probability of Lesion", line=2.2, cex=1)
abline(h=cutoff, lty=2)
legend('bottomleft', legend=c('Normal','Adenoma','Cancer'), pt.bg=c('royalblue1','orange','red'), pch=21, bty='n')
mtext('B', at=-0.5, font=2, side=3)
dev.off()

### Model stats for samples negative for FIT
neg_les_mod <- all_probs[all_data$dx!='normal' & all_data$fit_result < 100,2]
neg_ade_mod <- all_probs[all_data$dx=='adenoma' & all_data$fit_result < 100,2]
neg_canc_mod <- all_probs[all_data$dx=='cancer' & all_data$fit_result < 100,2]
neg_norm_mod <-  all_probs[all_data$dx=='normal' & all_data$fit_result < 100,2]

neg_les_mod_sens <- length(neg_les_mod[neg_les_mod>=cutoff]) / length(neg_les_mod)
neg_ade_mod_sens <- length(neg_ade_mod[neg_ade_mod>=cutoff]) / length(neg_ade_mod)
neg_canc_mod_sens <- length(neg_canc_mod[neg_canc_mod>=cutoff]) / length(neg_canc_mod)
neg_mod_spec <- length(neg_norm_mod[neg_norm_mod<cutoff]) / length(neg_norm_mod)
neg_mod_acc <- (length(neg_norm_mod[neg_norm_mod<cutoff]) + length(neg_les_mod[neg_les_mod>=cutoff])) / (length(neg_les_mod) + length(neg_norm_mod))
```

```{r figure4_barplot, echo=F, fig.width=9, fig.height=6, cache=T, results='hide'}
mmt_spec <- coords(les_roc, x=cutoff, input='thr', ret='spec')
fit_thr <- coords(les_fit_roc, x=mmt_spec, input='spec', ret='thr')
if(is.na(fit_thr)){fit_thr <- 7}
low_fit_spec <- coords(les_fit_roc, x=fit_thr, input='thr', ret='spec')


all_les_probs <- all_probs[all_data$dx!='normal',2]
nonAdv_probs <- all_probs[all_data$Dx_Bin=='Adenoma',2]
advAde_probs <- all_probs[all_data$Dx_Bin=='adv Adenoma',2]
all_ade_probs <- all_probs[all_data$dx=='adenoma',2]
s1_probs <- all_probs[all_data$stage==1,2]
s2_probs <- all_probs[all_data$stage==2,2]
s3_probs <- all_probs[all_data$stage==3,2]
s4_probs <- all_probs[all_data$stage==4,2]
all_canc_probs <- all_probs[all_data$dx=='cancer',2]

all_les_sens <- sum(all_les_probs>=cutoff) / length(all_les_probs)
nonAdv_sens <- sum(nonAdv_probs>=cutoff) / length(nonAdv_probs)
advAde_sens <- sum(advAde_probs>=cutoff) / length(advAde_probs)
all_ade_sens <- sum(all_ade_probs>=cutoff) / length(all_ade_probs)
s1_sens <- sum(s1_probs>=cutoff) / length(s1_probs)
s2_sens <- sum(s2_probs>=cutoff) / length(s2_probs)
s3_sens <- sum(s3_probs>=cutoff) / length(s3_probs)
s4_sens <- sum(s4_probs>=cutoff) / length(s4_probs)
all_canc_sens <- sum(all_canc_probs>=cutoff) / length(all_canc_probs)

all_les_fit <- all_data[all_data$dx!='normal','fit_result']
nonAdv_fit <- all_data[all_data$Dx_Bin=='Adenoma','fit_result']
advAde_fit <- all_data[all_data$Dx_Bin=='adv Adenoma','fit_result']
all_ade_fit <- all_data[all_data$dx=='adenoma','fit_result']
s1_fit <- all_data[all_data$stage==1,'fit_result']
s2_fit <- all_data[all_data$stage==2,'fit_result']
s3_fit <- all_data[all_data$stage==3,'fit_result']
s4_fit <- all_data[all_data$stage==4,'fit_result']
all_canc_fit <- all_data[all_data$dx=='cancer','fit_result']

all_les_fit_sens <- sum(all_les_fit>=fit_thr) / length(all_les_fit)
nonAdv_fit_sens <- sum(nonAdv_fit>=fit_thr) / length(nonAdv_fit)
advAde_fit_sens <- sum(advAde_fit>=fit_thr) / length(advAde_fit)
all_ade_fit_sens <- sum(all_ade_fit>=fit_thr) / length(all_ade_fit)
s1_fit_sens <- sum(s1_fit>=fit_thr) / length(s1_fit)
s2_fit_sens <- sum(s2_fit>=fit_thr) / length(s2_fit)
s3_fit_sens <- sum(s3_fit>=fit_thr) / length(s3_fit)
s4_fit_sens <- sum(s4_fit>=fit_thr) / length(s4_fit)
all_canc_fit_sens <- sum(all_canc_fit>=fit_thr) / length(all_canc_fit)


sens_matrix <- matrix(c(all_les_fit_sens,all_les_sens,nonAdv_fit_sens,nonAdv_sens,advAde_fit_sens,advAde_sens,all_ade_fit_sens,all_ade_sens,s1_fit_sens,s1_sens,s2_fit_sens,s2_sens,s3_fit_sens,s3_sens,s4_fit_sens,s4_sens,all_canc_fit_sens,all_canc_sens), byrow=F, nrow=2)
rownames(sens_matrix) <- c('FIT','MMT')


pdf('results/figures/Figure4.pdf', height=6, width=10)
par(mar=c(3.5,4.5,0,0))
x_labels <- c(sprintf('All\nLesions\n(n=%i)',length(all_les_probs)),sprintf('Nonadvanced\nAdenomas\n(n=%i)',length(nonAdv_probs)),sprintf('Advanced\nAdenomas\n(n=%i)',length(advAde_probs)),sprintf('All\nAdenomas\n(n=%i)',length(all_ade_probs)),sprintf('Cancer\nStage I\n(n=%i)',length(s1_probs)), sprintf('Cancer\nStage II\n(n=%i)',length(s2_probs)), sprintf('Cancer\nStage III\n(n=%i)',length(s3_probs)), sprintf('Cancer\nStage IV\n(n=%i)',length(s4_probs)),sprintf('All\nCancers\n(n=%i)',length(all_canc_probs)))
bp<-barplot(sens_matrix, beside=T, ylab='Sensitvity (%)', ylim=c(0,1.1), axisnames=F, col=c('grey90','grey25'),axis.lty=1, yaxt='n')
axis(2, at=seq(0,1,0.1), labels=seq(0,100,10))
mtext(text=x_labels, side=1, at=c(2,5,8,11,14,17,20,23,26), line=2, cex=0.8, font=c(2,1,1,2,1,1,1,1,2))
legend(0,1, legend=c(sprintf('FIT (at %.1f%% specificity)',mmt_spec*100),sprintf('MMT (at %.1f%% specificity)',mmt_spec*100)), col=c('grey90','grey25'), pch=15, bty='n', pt.cex=2)


### Significance tests with pROC
les_p <- roc.test(les_fit_roc, les_roc, specificity=mmt_spec, method='specificity', alternative='less', paired=TRUE, boot.n=1000)$p.value

nonAdv_fit_roc <- roc(all_data[all_data$Dx_Bin=='Adenoma' | all_data$dx=='normal', 'lesion']~all_data[all_data$Dx_Bin=='Adenoma' | all_data$dx=='normal', 'fit_result'])
nonAdv_roc <- roc(all_data[all_data$Dx_Bin=='Adenoma' | all_data$dx=='normal', 'lesion']~all_probs[all_data$Dx_Bin=='Adenoma' | all_data$dx=='normal', 2])
nonAdv_p <- roc.test(nonAdv_fit_roc,nonAdv_roc, specificity=mmt_spec, method='specificity', alternative='less', paired=TRUE, boot.n=1000)$p.value

advAde_fit_roc <- roc(all_data[all_data$Dx_Bin=='adv Adenoma' | all_data$dx=='normal', 'lesion']~all_data[all_data$Dx_Bin=='adv Adenoma' | all_data$dx=='normal', 'fit_result'])
advAde_roc <- roc(all_data[all_data$Dx_Bin=='adv Adenoma' | all_data$dx=='normal', 'lesion']~all_probs[all_data$Dx_Bin=='adv Adenoma' | all_data$dx=='normal', 2])
advAde_p <- roc.test(advAde_fit_roc, advAde_roc, specificity=mmt_spec, method='specificity', alternative='less', paired=TRUE, boot.n=1000)$p.value


allAde_p <- roc.test(ade_fit_roc, ade_roc, specificity=mmt_spec, method='specificity', alternative='less', paired=TRUE, boot.n=1000)$p.value

s1_fit_roc <- roc(all_data[all_data$stage==1 | all_data$dx=='normal', 'lesion']~all_data[all_data$stage==1 | all_data$dx=='normal', 'fit_result'])
s1_roc <- roc(all_data[all_data$stage==1 | all_data$dx=='normal', 'lesion']~all_probs[all_data$stage==1 | all_data$dx=='normal', 2])
s1_p <- roc.test(s1_fit_roc, s1_roc, specificity=mmt_spec, method='specificity', alternative='less', paired=TRUE, boot.n=1000)$p.value
  
s2_fit_roc <- roc(all_data[all_data$stage==2 | all_data$dx=='normal', 'lesion']~all_data[all_data$stage==2 | all_data$dx=='normal', 'fit_result']) 
s2_roc <- roc(all_data[all_data$stage==2 | all_data$dx=='normal', 'lesion']~all_probs[all_data$stage==2 | all_data$dx=='normal', 2])
s2_p <- roc.test(s2_fit_roc, s2_roc, specificity=mmt_spec, method='specificity', alternative='less', paired=TRUE, boot.n=1000)$p.value

s3_fit_roc <- roc(all_data[all_data$stage==3 | all_data$dx=='normal', 'lesion']~all_data[all_data$stage==3 | all_data$dx=='normal', 'fit_result'])
s3_roc <- roc(all_data[all_data$stage==3 | all_data$dx=='normal', 'lesion']~all_probs[all_data$stage==3 | all_data$dx=='normal', 2])
s3_p <- roc.test(s3_fit_roc, s3_roc, specificity=mmt_spec, method='specificity', alternative='less', paired=TRUE, boot.n=1000)$p.value

s4_fit_roc <- roc(all_data[all_data$stage==4 | all_data$dx=='normal', 'lesion']~all_data[all_data$stage==4 | all_data$dx=='normal', 'fit_result'])
s4_roc <- roc(all_data[all_data$stage==4 | all_data$dx=='normal', 'lesion']~all_probs[all_data$stage==4 | all_data$dx=='normal', 2])
s4_p <- roc.test(s4_fit_roc, s4_roc, specificity=mmt_spec, method='specificity', alternative='less', paired=TRUE, boot.n=1000)$p.value

allCanc_p <- roc.test(canc_fit_roc, canc_roc, specificity=mmt_spec, method='specificity', alternative='less', paired=TRUE, boot.n=1000)$p.value


all_p <- c(les_p, nonAdv_p, advAde_p, allAde_p, s1_p, s2_p, s3_p, s4_p, allCanc_p)
for(i in 1:length(all_p)){
  if(all_p[i]<0.1){
    text(x=i*3-1, y=sens_matrix[2,i]*1.1, '*', cex=2.5)  
    #text(x=i*3-1, y=sens_matrix[2,i]*1.05, sprintf('p=%.1g',all_p[i]))
  }
}


dev.off()

```

```{r table1_sensSpec, echo=F, cache=T}
#Numbers of each diagnosis by colonoscopy
les_mod <- all_probs[all_data$dx!='normal',2]
advAde_mod <- all_probs[all_data$Dx_Bin=='adv Adenoma',2]
nonAdv_mod <- all_probs[all_data$Dx_Bin=='Adenoma',2]
ade_mod <- all_probs[all_data$dx=='adenoma',2]
canc_mod <- all_probs[all_data$dx=='cancer',2]
norm_mod <- all_probs[all_data$dx=='normal',2]
num_les <- length(les_mod)
num_advAde <- length(advAde_mod)
num_nonAdv <- length(nonAdv_mod)
num_ade <- length(ade_mod)
num_canc <- length(canc_mod)
num_norm <- length(norm_mod)

#Numbers of diagnoses observed by our model
les_obs <- length(les_mod[les_mod>=cutoff])
advAde_obs <- length(advAde_mod[advAde_mod>=cutoff])
nonAdv_obs <- length(nonAdv_mod[nonAdv_mod>=cutoff])
ade_obs <- length(ade_mod[ade_mod>=cutoff])
canc_obs <- length(canc_mod[canc_mod>=cutoff])
norm_obs <- length(norm_mod[norm_mod<cutoff])

# sensitivity and 95% CI calculations for model
les_sens_ci <- ci.coords(les_roc, x=cutoff, input='threshold', ret='sensitivity')
les_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*les_sens_ci[2],100*les_sens_ci[1],100*les_sens_ci[3])

advAde_roc <- roc(les_data$lesion[all_data$Dx_Bin=='adv Adenoma' | all_data$Dx_Bin=='Normal'] ~ all_probs[all_data$Dx_Bin=='adv Adenoma' | all_data$Dx_Bin=='Normal',2])
nonAdv_roc <- roc(les_data$lesion[all_data$Dx_Bin=='Adenoma' | all_data$Dx_Bin=='Normal'] ~ all_probs[all_data$Dx_Bin=='Adenoma' | all_data$Dx_Bin=='Normal',2])

advAde_sens_ci<-ci.coords(advAde_roc, x=cutoff, input='threshold', ret='sensitivity')
advAde_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*advAde_sens_ci[2],100*advAde_sens_ci[1],100*advAde_sens_ci[3])

nonAdv_sens_ci<-ci.coords(nonAdv_roc, x=cutoff, input='threshold', ret='sensitivity')
nonAdv_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*nonAdv_sens_ci[2],100*nonAdv_sens_ci[1],100*nonAdv_sens_ci[3])

ade_sens_ci<-ci.coords(ade_roc, x=cutoff, input='threshold', ret='sensitivity')
ade_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*ade_sens_ci[2],100*ade_sens_ci[1],100*ade_sens_ci[3])

canc_sens_ci<-ci.coords(canc_roc, x=cutoff, input='threshold', ret='sensitivity')
canc_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*canc_sens_ci[2],100*canc_sens_ci[1],100*canc_sens_ci[3])

mod_spec_ci<-ci.coords(les_roc, x=cutoff, input='threshold', ret='specificity')
mod_spec <- sprintf('%.3g (%.3g-%.3g)',100*mod_spec_ci[2],100*mod_spec_ci[1],100*mod_spec_ci[3])


#Numbers of diagnoses observed by FIT
les_fit <- all_data[all_data$dx!='normal','fit_result']
advAde_fit <- all_data[all_data$Dx_Bin=='adv Adenoma','fit_result']
nonAdv_fit <- all_data[all_data$Dx_Bin=='Adenoma','fit_result']
ade_fit <- all_data[all_data$dx=='adenoma','fit_result']
canc_fit <- all_data[all_data$dx=='cancer','fit_result']
norm_fit <- all_data[all_data$dx=='normal','fit_result']
les_fit_obs <- length(les_fit[les_fit>=100])
advAde_fit_obs <- length(advAde_fit[advAde_fit>=100])
nonAdv_fit_obs <- length(nonAdv_fit[nonAdv_fit>=100])
ade_fit_obs <- length(ade_fit[ade_fit>=100])
canc_fit_obs <- length(canc_fit[canc_fit>=100])
norm_fit_obs <- length(norm_fit[norm_fit<100])

# sensitivity and 95% CI calculations for FIT
les_fit_sens_ci <- ci.coords(les_fit_roc, x=100, input='threshold', ret='sensitivity')
les_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*les_fit_sens_ci[2],100*les_fit_sens_ci[1],100*les_fit_sens_ci[3])

advAde_fit_roc <- roc(les_data$lesion[all_data$Dx_Bin=='adv Adenoma' | all_data$Dx_Bin=='Normal'] ~ all_data[all_data$Dx_Bin=='adv Adenoma' | all_data$Dx_Bin=='Normal', 'fit_result'])
nonAdv_fit_roc <- roc(les_data$lesion[all_data$Dx_Bin=='Adenoma' | all_data$Dx_Bin=='Normal'] ~ all_data[all_data$Dx_Bin=='Adenoma' | all_data$Dx_Bin=='Normal', 'fit_result'])

advAde_fit_sens_ci<-ci.coords(advAde_fit_roc, x=100, input='threshold', ret='sensitivity')
advAde_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*advAde_fit_sens_ci[2],100*advAde_fit_sens_ci[1],100*advAde_fit_sens_ci[3])

nonAdv_fit_sens_ci<-ci.coords(nonAdv_fit_roc, x=100, input='threshold', ret='sensitivity')
nonAdv_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*nonAdv_fit_sens_ci[2],100*nonAdv_fit_sens_ci[1],100*nonAdv_fit_sens_ci[3])

ade_fit_sens_ci<-ci.coords(ade_fit_roc, x=100, input='threshold', ret='sensitivity')
ade_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*ade_fit_sens_ci[2],100*ade_fit_sens_ci[1],100*ade_fit_sens_ci[3])

canc_fit_sens_ci<-ci.coords(canc_fit_roc, x=100, input='threshold', ret='sensitivity')
canc_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*canc_fit_sens_ci[2],100*canc_fit_sens_ci[1],100*canc_fit_sens_ci[3])

fit_spec_ci<-ci.coords(les_fit_roc, x=100, input='threshold', ret='specificity')
fit_spec <- sprintf('%.3g (%.3g-%.3g)',100*fit_spec_ci[2],100*fit_spec_ci[1],100*fit_spec_ci[3])

#Build Output table

stats_table <- matrix(NA, ncol=6, nrow=7)
colnames(stats_table) <- c('Most Advanced Finding','','FIT','','MMT','')
stats_table[1,] <- c('','','Positive Results','Sensitivity (95% CI)','Positive Results','Sensitivity (95% CI)')
stats_table[2,] <- c('','no.','no.','%','no.','%')
stats_table[3,] <- c('Cancer', num_canc, canc_fit_obs, canc_fit_sens, canc_obs, canc_mod_sens)
stats_table[4,] <- c('Adenoma', num_ade, ade_fit_obs, ade_fit_sens, ade_obs, ade_mod_sens)
stats_table[5,] <- c('Any Lesion', num_les, les_fit_obs, les_fit_sens, les_obs, les_mod_sens)
stats_table[6,] <- c('','','Negative Results','Specificity (95% CI)','Negative Results','Specificity (95% CI)')
stats_table[7,] <- c('Negative Colonosopies', num_norm, norm_fit_obs, fit_spec, norm_obs, mod_spec)

write(kable(stats_table), file='results/tables/table1.md')

```

```{r table2_screeningPop, echo=F}
mmt_spec <- coords(les_roc, x=cutoff, input='thr', ret='spec')
fit_thr <- coords(les_fit_roc, x=mmt_spec, input='spec', ret='thr')
if(is.na(fit_thr)){fit_thr <- 100}
low_fit_spec <- coords(les_fit_roc, x=fit_thr, input='thr', ret='spec')


all_les_probs <- all_probs[all_data$dx!='normal',2]
nonAdv_probs <- all_probs[all_data$Dx_Bin=='Adenoma',2]
advAde_probs <- all_probs[all_data$Dx_Bin=='adv Adenoma',2]
all_ade_probs <- all_probs[all_data$dx=='adenoma',2]
s1_probs <- all_probs[all_data$stage==1,2]
s2_probs <- all_probs[all_data$stage==2,2]
s3_probs <- all_probs[all_data$stage==3,2]
s4_probs <- all_probs[all_data$stage==4,2]
all_canc_probs <- all_probs[all_data$dx=='cancer',2]

all_les_sens <- sum(all_les_probs>=cutoff) / length(all_les_probs)
nonAdv_sens <- sum(nonAdv_probs>=cutoff) / length(nonAdv_probs)
advAde_sens <- sum(advAde_probs>=cutoff) / length(advAde_probs)
all_ade_sens <- sum(all_ade_probs>=cutoff) / length(all_ade_probs)
s1_sens <- sum(s1_probs>=cutoff) / length(s1_probs)
s2_sens <- sum(s2_probs>=cutoff) / length(s2_probs)
s3_sens <- sum(s3_probs>=cutoff) / length(s3_probs)
s4_sens <- sum(s4_probs>=cutoff) / length(s4_probs)
all_canc_sens <- sum(all_canc_probs>=cutoff) / length(all_canc_probs)

all_les_fit <- all_data[all_data$dx!='normal','fit_result']
nonAdv_fit <- all_data[all_data$Dx_Bin=='Adenoma','fit_result']
advAde_fit <- all_data[all_data$Dx_Bin=='adv Adenoma','fit_result']
all_ade_fit <- all_data[all_data$dx=='adenoma','fit_result']
s1_fit <- all_data[all_data$stage==1,'fit_result']
s2_fit <- all_data[all_data$stage==2,'fit_result']
s3_fit <- all_data[all_data$stage==3,'fit_result']
s4_fit <- all_data[all_data$stage==4,'fit_result']
all_canc_fit <- all_data[all_data$dx=='cancer','fit_result']

all_les_fit_sens <- sum(all_les_fit>=fit_thr) / length(all_les_fit)
nonAdv_fit_sens <- sum(nonAdv_fit>=fit_thr) / length(nonAdv_fit)
advAde_fit_sens <- sum(advAde_fit>=fit_thr) / length(advAde_fit)
all_ade_fit_sens <- sum(all_ade_fit>=fit_thr) / length(all_ade_fit)
s1_fit_sens <- sum(s1_fit>=fit_thr) / length(s1_fit)
s2_fit_sens <- sum(s2_fit>=fit_thr) / length(s2_fit)
s3_fit_sens <- sum(s3_fit>=fit_thr) / length(s3_fit)
s4_fit_sens <- sum(s4_fit>=fit_thr) / length(s4_fit)
all_canc_fit_sens <- sum(all_canc_fit>=fit_thr) / length(all_canc_fit)


total_pop <- 80494283
canc_pop <- round(total_pop*0.003, 0)
advAde_pop <- round(total_pop*0.057, 0)
nonAdv_pop <- round(total_pop*0.177, 0)
norm_pop <- total_pop-(canc_pop+advAde_pop+nonAdv_pop)

canc_tp_mmt <- round(canc_pop*all_canc_sens, 0)
canc_tp_fit <- round(canc_pop*all_canc_fit_sens, 0)
advAde_tp_mmt <- round(advAde_pop*advAde_sens, 0)
advAde_tp_fit <- round(advAde_pop*advAde_fit_sens, 0)
nonAdv_tp_mmt <- round(nonAdv_pop*nonAdv_sens, 0)
nonAdv_tp_fit <- round(nonAdv_pop*nonAdv_fit_sens, 0)

fp_mmt <- norm_pop*(1-mmt_spec)
fp_fit <-norm_pop*(1-fit_spec_ci[2])

truePos_table <- matrix(c('Cancer','Advanced Adenoma','Non-advanced Adenoma','0.03%', '5.7%', '17.7%', canc_pop, advAde_pop, nonAdv_pop, canc_tp_fit, advAde_tp_fit, nonAdv_tp_fit, canc_tp_mmt, advAde_tp_mmt, nonAdv_tp_mmt), nrow=3)
colnames(truePos_table) <- c('Condition','Prevalence','Number of Persons, ages 50-75 years, with Condition','True Positives identified by FIT','True Positives identified by MMT')

write(kable(truePos_table), file='results/tables/table2.md')
```


```{r figS6_gender, echo=F, results='hide', cache=T, fig.width=9, fig.height=6}
m_roc <- roc(all_data$lesion[all_data$Gender == 'm'] ~ all_probs[all_data$Gender == 'm',2])
f_roc <- roc(all_data$lesion[all_data$Gender == 'f'] ~ all_probs[all_data$Gender == 'f',2])
gender_pval <- roc.test(m_roc,f_roc)$p.value


pdf('results/figures/figS6.pdf', height=6, width=9)
layout(matrix(c(1,2),nrow=1), widths=c(8,5))
par(mar=c(4.5,4.5,1,1))

plot(c(1,0),c(0,1), type='l', lty=3, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='Sensitivity', xlab='Specificity')
plot(les_roc, col='purple', lwd=2, add=T)
plot(m_roc, col='blue', lwd=2, add=T)
plot(f_roc, col='palevioletred1', lwd=2, add=T)

points(coords(les_roc, x=cutoff, input='threshold', ret='specificity'),coords(les_roc, x=cutoff, input='threshold', ret='sensitivity'), col='purple', cex=1.5, pch=16)
points(coords(m_roc, x=cutoff, input='threshold', ret='specificity'),coords(m_roc, x=cutoff, input='threshold', ret='sensitivity'), col='blue', cex=1.5, pch=16)
points(coords(f_roc, x=cutoff, input='threshold', ret='specificity'),coords(f_roc, x=cutoff, input='threshold', ret='sensitivity'), col='palevioletred1', cex=1.5, pch=16)

legend('bottomright', legend=c(sprintf('All Subjects: %.3g',les_roc$auc),
                               sprintf('Males: %.3g',m_roc$auc),
                               sprintf('Females: %.3g',f_roc$auc),
                               sprintf('Male vs Female: p=%.3g',gender_pval)
                               ),lty=c(1,1,1,0), lwd=2, col=c('purple','blue','palevioletred1'), bty='n')
mtext('A', at=1.12, font=2, side=3)

plot(1, type="n", xlim=c(0.5,7), ylim=c(0,1.05), ylab="MMT Probability of Lesion", xlab="", xaxt="n")
stripchart(at=1, all_probs[all_data$dx=='normal' & all_data$Gender == 'm', 2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='royalblue1', add=T)
stripchart(at=2, all_probs[all_data$dx=='adenoma' & all_data$Gender == 'm',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='orange', add=T)
stripchart(at=3, all_probs[all_data$dx=='cancer' & all_data$Gender == 'm',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)

stripchart(at=4.5, all_probs[all_data$dx=='normal' & all_data$Gender == 'f',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='royalblue1', add=T)
stripchart(at=5.5, all_probs[all_data$dx=='adenoma' & all_data$Gender == 'f',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='orange', add=T)
stripchart(at=6.6, all_probs[all_data$dx=='cancer' & all_data$Gender == 'f',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)

axis(1, at=c(2,5), labels=c('Males','Females'))
abline(h=cutoff, lty=2)
legend('topleft', legend=c('Normal','Adenoma','Cancer'), pt.bg=c('royalblue1','orange','red'), pch=21, bty='n', pt.cex=1, horiz=T, cex=0.75)
mtext('B', at=-1, font=2, side=3)
dev.off()

m_sens <- coords(m_roc, x=cutoff, input='threshold', ret='sensitivity')
m_spec <- coords(m_roc, x=cutoff, input='threshold', ret='specificity')
f_sens <- coords(f_roc, x=cutoff, input='threshold', ret='sensitivity')
f_spec <- coords(f_roc, x=cutoff, input='threshold', ret='specificity')

m_ade_probs <- all_probs[all_data$Gender=='m' & all_data$dx=='adenoma', 2]
f_ade_probs <- all_probs[all_data$Gender=='f' & all_data$dx=='adenoma', 2]

m_ade_sens <- length(m_ade_probs[m_ade_probs > cutoff])/length(m_ade_probs)
f_ade_sens <- length(f_ade_probs[f_ade_probs > cutoff])/length(f_ade_probs)

m_canc_probs <- all_probs[all_data$Gender=='m' & all_data$dx=='cancer', 2]
f_canc_probs <- all_probs[all_data$Gender=='f' & all_data$dx=='cancer', 2]

m_canc_sens <- length(m_canc_probs[m_canc_probs > cutoff])/length(m_canc_probs)
f_canc_sens <- length(f_canc_probs[f_canc_probs > cutoff])/length(f_canc_probs)

#correct for diagnosis
fit_aov <- aov(meta$fit_result~meta$Gender*meta$dx)
fit_aov_p <- summary(fit_aov)[[1]][["Pr(>F)"]][1]

dist <- read.table('data/glne007.final.an.unique_list.0.03.subsample.0.03.filter.thetayc.0.03.square.dist', skip=1)
mod_adonis <- adonis(dist~meta$Gender*meta$dx)
mod_adonis_p <- mod_adonis[[1]][["Pr(>F)"]][1]

```

```{r figS7_metadata_model, echo=F, cache=T, results='hide'}

demo_data <- all_data[,c(les_model$Xopt,'Age','BMI','Gender','Smoke')]
demo_data$lesion <- les_data$lesion
rows <- apply(is.na(demo_data), 1, sum)==0
demo_data <- demo_data[rows,]

set.seed(010516)
demo_rf <-AUCRF(lesion ~ ., data=demo_data, ntree=500, pdel=0.99, ranking='MDG')
demo_probs <- predict(demo_rf$RFopt,type='prob')[,2]
demo_roc <- roc(demo_data$lesion~demo_probs)

mmt_probs <- all_probs[rows,2]
mmt_roc <- roc(demo_data$lesion~mmt_probs)

pdf('results/figures/figS7.pdf', height=5, width=6)
par(mar=c(4.5,4.5,1,1))
plot(c(1,0),c(0,1), type='l', lty=3, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='', xlab='', cex.axis=1.2)
plot(mmt_roc, col='red', lwd=2, add=T)
plot(demo_roc, col='green3', lwd=2, add=T)
plot(les_fit_roc, col='blue', lwd=2, add=T, lty=1)
mtext(side=2, text="Sensitivity", line=2.5, cex=1)
mtext(side=1, text="Specificity", line=2.5, cex=1)
legend('bottomright', legend=c(sprintf('MMT+Age,BMI,Gender,Smoking: %.3g', demo_roc$auc),
                                sprintf('MMT: %.3g',mmt_roc$auc),
                               sprintf('FIT: %.3g',les_fit_roc$auc)),
                              lty=1, lwd=2, col=c('green3','red','blue'), bty='n', cex=0.8, seg.len=3)
dev.off()

mmt_sens <- coords(mmt_roc, x=cutoff, input='thr', ret='sens')
demo_sens <- coords(demo_roc, x=mmt_spec, input='spec', ret='sens')
demo_cutoff <- coords(demo_roc, x=demo_sens, input='sens', ret='thr')

mc <- matrix(c(sum(mmt_probs>=cutoff & demo_probs>=demo_cutoff),
               sum(mmt_probs>=cutoff & demo_probs<demo_cutoff),
               sum(mmt_probs<cutoff & demo_probs>=demo_cutoff),
               sum(mmt_probs<cutoff & demo_probs<demo_cutoff)), nrow=2)

demo_mc <- mcnemar.test(mc)

```

```{r figures, echo=F}

canc_ROC <- 'Fig. 1C'
canc_scatter <- 'Fig. 1D'
ade_ROC <- 'Fig. 1A'
ade_scatter <- 'Fig. 1B'
ROC <- 'Fig. 2A'
stripFIT <- 'Fig. 2B'
stripMMT <- 'Fig. 2C'
table1 <- 'Table 1'
scatter <- 'Fig. 3A'
fit_neg_strip <- 'Fig. 3B'
bar <- 'Fig. 4'
pred_values <- 'Table 2'

ade_aucrf <- 'Fig. S1A'
ade_varImp <- 'Fig. S1B'
ade_cv_rocs <- 'Fig. S2A'
canc_cv_rocs <- 'Fig. S2B'
canc_aucrf <- 'Fig. S3A'
canc_varImp <- 'Fig. S3B'
canc_stripChart <- 'Fig. S3C'
mmt_otus <- 'Fig. S4'
mmt_cv_rocs <- 'Fig. S5'
gender <- 'Fig. S6'
demo_fig <- 'Fig. S7'


```


###Abstract  
**Background**  
Colorectal cancer is the second leading cause of death among cancers in the United States. Although individuals diagnosed early have a greater than 90% chance of survival, more than one-third of individuals do not adhere to screening recommendations partly because the standard diagnostics, colonoscopy and sigmoidoscopy, are expensive and invasive. Thus, there is a great need to improve the sensitivity of non-invasive tests to detect early stage cancers and adenomas. Numerous studies have identified shifts in the composition of the gut microbiota associated with the progression of colorectal cancer, suggesting that the gut microbiota may represent a reservoir of biomarkers that would complement existing non-invasive methods such as the widely used fecal immunochemical test (FIT). 

**Methods**  
We sequenced the 16S rRNA genes from the stool samples of 490 patients. We used the relative abundances of the bacterial populations within each sample to develop a random forest classification model that detects colonic lesions using the relative abundance of gut microbiota and the concentration of hemoglobin in stool.

**Results**  
The microbiota-based random forest model detected `r sprintf('%.1f', 100*canc_sens_ci[2])`% of cancers and `r sprintf('%.1f', 100*ade_sens_ci[2])`% of adenomas while FIT alone detected `r sprintf('%.1f', 100*canc_fit_sens_ci[2])`% and `r sprintf('%.1f', 100*ade_fit_sens_ci[2])`%, respectively. Of the colonic lesions missed by FIT, the model detected `r sprintf('%.1f', 100*neg_canc_mod_sens)`% of cancers and `r sprintf('%.1f', 100*neg_ade_mod_sens)`% of adenomas. We confirmed known assocaitions of _Porphyromonas assaccharolytica_, _Peptostreptococcus stomatis_, _Parvimonas micra_, and _Fusobacterium nucleatum_ with CRC. Yet, we found that the loss of potentially beneficial organisms, such as members of the Lachnospiraceae, was more predictive for identifying patients with adenomas when used in combination with FIT.

**Conclusions**  
These findings demonstrate the potential for microbiota analysis to complement existing screening methods to improve detection of colonic lesions.

###Background  
Colorectal cancer mortality has steadily declined in recent decades, due in large part to increased screening [@siegel2014colorectal]. Yet current screening tests, the fecal immunochemical test (FIT) and the multitarget DNA test, have a sensitivity of 7.6% and 17.2%, respectively, for detecting non-advanced adenoma – just the type of early lesion that screening is meant to identify [@imperiale2014multitarget]. Although structural exams including colonoscopy and sigmoidoscopy are able to detect both adenomas and carcinomas, the high cost and invasive nature are barriers for many people. Fear, discomfort, and embarrassment are among the most cited reasons patients choose to forego CRC screening [@jones2010barriers]. Likewise the large disparity in screening rates between those with and without health insurance highlights the need for inexpensive screening methods [@hsia2000importance;@centers2013vital;@siegel2014colorectal]. Unfortunately cheaper, less invasive stool-based tests like guaic fecal occult blood test and FIT are unable to reliably detect adenomas [@hundt2009comparative]. The newly introduced stool DNA panel has improved accuracy compared to FIT, but is still limited in its ability to accurately detect adenomas [@imperiale2014multitarget]. Thus there is need for novel screening methods that are inexpensive and capable of detecting both cancer and adenomas.

The gut microbiota, the collection of microorganisms that inhabit the gastrointestinal tract, are one potential source of biomarkers for detecting colonic lesions. Numerous studies have observed alterations in the gut bacterial communities of patients with CRC [@wang2012structural;@chen2013decreased;@chen2012human;@shen2010molecular;@kostic2012genomic;@feng2015gut]. Experiments in animal models have demonstrated that such alterations have the potential to accelerate tumorigenesis [@zackularMbio]. Furthermore, several members of the gut microbiota have been shown to potentiate both the development and progression of CRC by a variety of mechanisms [@kostic2013fusobacterium;@wu2009human;@arthur2012intestinal]. Although each of these organisms may play a role in certain cases of CRC, none of them is present in every case. Therefore we postulate that no one organism is an effective biomarker on its own and that focusing on a single bacterial population excludes the potential that the microbial etiology of the disease is actually polymicrobial.

Two recent studies used statistical models that take into account the abundances of multiple bacterial species and the results of guaic fecal occult blood test (gFOBT) to distinguish healthy individuals from those with CRC [@zackular2014human;@zeller2014potential]. The analysis by Zackular et al. [@zackular2014human] used samples from a limited number of subjects (N=30 normal, 30 adenoma, and 30 carcinoma), while that of Zeller et al [@zeller2014potential] had a larger cohort from multiple clinical sites (N=156 and N=335). A shortcoming of the Zeller study was the pooling of subjects with non-advanced adenomas with control subjects as well as the exclusion of subjects with advanced adenomas. A limitation of both studies was that they relied on gFOBT rather than FIT to detect hemoglobin in stool. FIT provides a quantitatve measure of hemoglobin concentrations and has largely replaced gFOBT clinically because of its improved sensitivity. Regardless of their weaknesses, these studies demonstrated the feasibility of using microbiome data identify subjects with colonic lesions.

In the present study, we demonstrate the potential for microbiota analysis to complement FIT for improved detection of colonic lesions, especially adenomas. We utilized the random forest algorithm, which is a decision tree-based machine learning algorithm for classification that accounts for non-linear data and interactions among features and includes an internal cross-validation to prevent overfitting [@liaw2002classification]. With this method we identified bacterial populations that could distinguish healthy individuals from those with adenomas or carcinomas.  In doing so, we confirmed previously observed associations of certain bacterial taxa with CRC. Many lesions detected using the microbiota were distinct from those detected by FIT, suggesting the microbiota could complement FIT to improve sensitivity. By incorporating data on hemoglobin and bacterial abundances into a single model (labeled the Multitarget Microbiota Test or MMT), we were able to improve the sensitivity for adenomas and cancer compared to FIT alone.


###Methods

**Study Design/Patient sampling.** Eligible patients for this study were at least 18 years old, willing to sign informed consent, able to tolerate removal of 58 mL of blood, and willing to collect a stool sample. Patient age at the time of enrollment ranged from 29 to 89 with a median of 60. All patients were asymptomatic and were excluded if they had undergone surgery, radiation, or chemotherapy for current CRC prior to baseline samples or had inflammatory bowel disease, known hereditary non-polyposis CRC, or familial adenomatous polyposis. Colonoscopies were performed and fecal samples were collected from subjects in 4 locations: Toronto (Ontario, Canada), Boston (Massachusetts, USA), Houston (Texas, USA), and Ann Arbor (Michigan, USA). Patient diagnoses were determined by colonoscopic examination and histopathological review of any biopsies taken. Patients with an adenoma greater than 1cm, more than three adenomas of any size, or an adenoma with villous histology were classified as advanced adenoma. Whole evacuated stool was collected from each patient either prior to colonoscopy preparation or 1-2 weeks after colonoscopy. This has been shown to be sufficient time for the microbiota to recover from colonoscopy preparation [@OBrien2013].  Stool samples were packed in ice, shipped to a processing center via next day delivery and stored at -80˚C. The University of Michigan Institutional Review Board approved this study, and all subjects provided informed consent. This study conformed to the guidelines of the Helsinki Declaration.


**Fecal Immunochemical Tests.**  Fecal material for FIT was collected from frozen stool aliquots using OC FIT-CHEK sampling bottles (Polymedco Inc.) and processed using an OC-Auto Micro 80 automated system (Polymedco Inc.). Hemoglobin concentrations were used for generating ROC curves for FIT and for building the MMT.

**16S rRNA Gene Sequencing.**  DNA was extracted from approximately 50 mg of fecal material from each subject using the PowerSoil-htp 96 Well Soil DNA isolation kit (MO BIO Laboratories) and an epMotion 5075 automated pipetting system (Eppendorf). The V4 region of the bacterial 16S rRNA gene was amplified using custom barcoded primers and sequenced as described previously using an Illumina MiSeq sequencer [@miseq]. The 490 samples were divided into three sequencing runs to increase the per sample sequencing depth. Although the same percentage of samples from the three groups were represented on each sequencing run, samples were randomly assigned to the sequencing runs to avoid confounding our analysis based on diagnosis or demographics.

**Sequence Curation.**  The 16S rRNA gene sequences were curated using the mothur software package (v1.36), as described previously [@mothur;@miseq]. Briefly, paired-end reads were merged into contigs, screened for quality, aligned to SILVA 16S rRNA sequence database, and screened for chimeras. Sequences were classified using a naive Bayesian classifier trained against a 16S rRNA gene training set provided by the Ribosomal Database Project (RDP) [@wang2007naive]. Curated sequences were clustered into operational taxonomic units (OTUs) using a 97% similarity cutoff with the average neighbor clustering algorithm. Species-level classifications for OTUs of interest were determined by blasting the predominant sequences within each OTU to the NCBI 16S rRNA database. The putative species was only reported for OTUs with greater than 99% sequence identity to a single species in the database; otherwise the consensus RDP classification was used. The number of sequences in each sample was rarefied to 10,000 per sample to minimize the effects of uneven sampling. Only the 335 OTUs present in at least 5% of samples were included in the feature selection for the random forest models.

**Statistical Methods.**  All statistical analyses were performed using R (v.3.2.0). Random Forest models were generated using the AUCRF package [@calle2011auc]. All ROC curves presented for random forest models are based on the out-of-bag (OOB) error rates. For each model, leave-one-out and 10-fold cross-validations were performed to further estimate the generalization error of the model. The AUC of ROC curves were compared using the method described by DeLong et al. [@delong1988comparing]. The optimal cutoff for the MMT was determined using Youden's _J_ statistic [@youden1950index]. This cutoff was determined using the ROC curve for differentiating cancer from normal. Comparisons of sensitivities of FIT and the MMT at the same specificity were performed using the method deveoped by Pepe et al. with 1000 bootsrap replicates [@pepe2009estimation]. All of the aforementioned statistics for analyzing ROC curves were performed using the pROC package in R [@robin2011proc]. To control for diagnosis while testing the effects of sex on the microbiome we used PERMANOVA as implemented in the adonis function in the vegan R package [@dixon2003vegan].


###Results

_**Complementary detection of lesions by FIT and the microbiota.**_ 
We characterized the bacterial communities of stool samples from 490 patients using 16S rRNA gene sequencing.  Among these patients, 120 had CRC, 198 had adenomas, and 172 had no colonic lesions. In addition to characterizing the bacterial community, we tested each sample for the concentration of hemoglobin using FIT. With these data we compared the ability to detect lesions using FIT to using a microbiota-based model. First we developed a random forest classification model for differentiating healthy individuals from those with adenomas based on the relative abundance of bacterial populations in stool. We determined the optimal model using the AUC-RF algorithm for maximizing the area under the curve (AUC) of the receiver operating characteristic (ROC) curve for a random forest model [@calle2011auc]. The optimal model utilized 22 bacterial populations (`r ade_aucrf`). The vast majority of OTUs in the model (17 out of 22) belonged to the order Clostridales, 4 were associated with the genus _Bacteroides_, and one OTU was unclassified at the phylum level (`r ade_varImp`). The AUC for this and subsequent random forest models were generated based on the out-of-bag (OOB) probabilities for each sample. Additonal leave-one-out and 10-fold cross validations showed no significant difference in AUC compared to the OOB AUC (`r ade_cv_rocs`). The AUC for the microbiota model (`r signif(otu_ade_roc$auc, 3)`) was significantly different from a random assignment (p<0.001), but not significantly different from that of FIT (FIT AUC:`r signif(ade_fit_roc$auc, 3)`, p>0.05, `r ade_ROC`). At the 100 ng/ml cutoff FIT detected `r sprintf('%.3g',100*ade_fit_sens_ci[2])`% of adenomas with a specificity of `r sprintf('%.3g',100*fit_spec_ci[2])`%. Setting the microbiota model to the same `r sprintf('%.3g',100*otu_ade_spec)`% specificity resulted in `r sprintf('%.3g',100*otu_ade_sens)`% sensitivity for adenomas.  When comparing the results of the tests for each sample, only `r round(ade_table[4], 1)`% of adenomas were detected by both tests, while `r round(sum(ade_table[2:3]),1)`% were detected by only one of the two tests (`r ade_scatter`). Thus, the two tests detected small but distinct subsets of adenomas.

Next we generated a random forest model for differentiating normal individuals from those with cancer using the relative abundance of 34 bacterial populations (`r canc_aucrf`, `r canc_varImp`). Consistent with previous observations, the bacteria most strongly associated with CRC belonged to taxa commonly associated with periodontal disease [@warren2013co;@zeller2014potential;@yu2015metagenomic]. These include OTUs associated _Pophyromonas assaccharolytica_ (OTU105), _Fusobacterium nucleatum_ (OTU264), _Parvimonas micra_ (OTU281), _Peptostreptococcus stomatis_ (OTU310), _Gemella spp._ (OTU356), and an unclassified _Prevotella_ (OTU57)  (`r canc_stripChart`). The ROC curve for the model had an AUC of `r signif(otu_canc_roc$auc, 3)`, which was similar to AUCs reported for other microbiota-based models for CRC [@zackular2014human;@zeller2014potential]. The AUC of this model was significantly better than a random assignment (p<0.001), but was significantly lower than that of FIT (FIT AUC:`r signif(canc_fit_roc$auc, 3)`, `r pval(roc.test(otu_canc_roc,canc_fit_roc)$p.value)`, `r canc_ROC`). As with the adenoma versus normal model, we confirmed the OOB AUC with leave-one-out cross validation and 100 iterations of 10-fold cross validation (`r canc_cv_rocs`). At the manufacturer recommended cutoff of 100 ng/ml FIT detected `r sprintf('%.1f',100*canc_fit_sens_ci[2])`% of cancers with a specificity of `r sprintf('%.3g',100*fit_spec_ci[2])`%. At the same specificity the microbiota model detected `r sprintf('%.3g',100*otu_canc_sens)`% of cancers. Although more cancers were detected by FIT, the microbiota model was able to detect `r round(100*canc_table[3]/sum(canc_table[c(1,3)]), 1)`% of cancers missed by FIT (`r canc_scatter`). 


_**Multitarget Microbiota Test for colonic lesions.**_ 
Many of the adenomas and some of the carcinomas were detected by the microbiota models, but not FIT, suggesting that the two screening methods could complement each other if combined into a single test. Based on these observations, we developed a random forest model using both the microbiota and FIT that would differentiate normal individuals from those with any type of colonic lesion (i.e. adenoma or carcinoma). The optimal model, referred to as the Multitarget Microbiota Test (MMT), used the relative abundances of 23 OTUs and the concentration of hemoglobin as determined by FIT. Of those OTUs, 16 were members of the Firmicutes phylum, including 3 from the Ruminococcaceae family and 10 from the Lachnospiraceae family (`r mmt_otus`). Three OTUs were associated with the genus *Bacteroides*. The remaining OTUs were associated with *Porphyromonas*, *Parabacteroides*, *Collinsella*, and Enterobacteriaceae. The OTU associated with *Porphyromonas* was most closely related to *Porphyromonas asaccharolytica*, which has been previously shown to be predictive of CRC [@rex2009american;@zeller2014potential;@zackular2014human]. Interestingly the majority of OTUs used in the model, especially the Lachnospiraceae, were enriched in normal patients (`r mmt_otus`), suggesting that a loss of beneficial organisms in addition to the emergence of pathogens may be indicative of CRC development. As with the previous random forest models we performed leave-one-out cross validation and 100 iterations of 10-fold cross validation and found no difference in AUC compared to the OOB estimates (`r mmt_cv_rocs`).

_**Comparing MMT to FIT.**_ 
To determine whether microbiota sequence data could be used to complement FIT, we compared the performance of the MMT to FIT. For differentiating any lesions from normal, the AUC for the MMT was significantly higher than FIT (MMT AUC:`r signif(les_roc$auc, 3)`, FIT AUC:`r signif(les_fit_roc$auc, 3)`, `r pval(roc.test(les_roc, les_fit_roc)$p.value)`, `r ROC`). Subdividing the lesions, detecting adenomas by the MMT (AUC:`r signif(ade_roc$auc, 3)`) was significantly better than FIT (AUC:`r signif(ade_fit_roc$auc, 3)`, `r pval(roc.test(ade_roc, ade_fit_roc)$p.value)`), but not for differentiating cancer from normal (MMT AUC:`r signif(canc_roc$auc, 3)`, FIT AUC:`r signif(canc_fit_roc$auc, 3)`, `r pval(roc.test(canc_roc, canc_fit_roc)$p.value)`). To generate a categorical prediction from the MMT, we determined the models's optimal threshold for detecing cancer (`r round(cutoff, 3)` probabilitiy of a lesion) using Youden's J statisitc [@youden1950index]. Samples scoring above this cutoff were classified as lesions, and those below the cutoff were classified as normal. We then compared the sensitivity and specificity of the MMT to those of FIT using a threshold of 100 ng/ml of hemoglobin.  At these cutoffs the MMT detected `r sprintf('%.1f',100*canc_sens_ci[2])`% of cancers and `r sprintf('%.1f',100*ade_sens_ci[2])`% of adenomas compared to `r sprintf('%.1f',100*canc_fit_sens_ci[2])`% and `r sprintf('%.1f',100*ade_fit_sens_ci[2])`% for FIT (`r table1`, `r stripFIT`, `r stripMMT`). When adenomas and cancers were pooled together, the MMT detected `r sprintf('%.1f',100*les_sens_ci[2])`% of lesions, while FIT only detected `r sprintf('%.1f',100*les_fit_sens_ci[2])`%. However, the increased sensitivity of the MMT was accompanied by a decrease in specificity (`r sprintf('%.1f',100*mod_spec_ci[2])`%) compared to FIT (`r sprintf('%.1f',100*fit_spec_ci[2])`%).

To better understand the relationship between the MMT and FIT, we compared the results of the two tests for each sample (`r scatter`). All but one of the samples that tested positive by FIT also tested positive by the MMT. However the MMT was able to detect `r sprintf('%.1f',100*neg_canc_mod_sens)`% of cancers and `r signif(100*neg_ade_mod_sens, 3)`% of adenomas that FIT had failed to detect, while maintaining a specificity of `r signif(100*neg_mod_spec, 3)`% (`r fit_neg_strip`). This result demonstrated that incorporation of data from a subject's microbiota could complement FIT to improve its sensitivity.

To make a fairer comparison of the sensitivities of these two tests, we reduced the cutoff for FIT to 7 ng/ml to match the `r round(100*mmt_spec, 1)`% specificity of the MMT. At the lower cutoff for FIT there was no significant difference in sensitivity for cancer between the two tests (`r pval(allCanc_p)`), but the MMT remained significantly more sensitive for detecting adenomas (`r pval(allAde_p)`) and all lesions grouped together (`r pval(les_p)`, `r bar`). 

The purpose of screening is to identify asymptomatic individuals with early stage disease (i.e., true positives).  Therefore, we estimated the number of true positives captured through FIT and MMT in the recommended screening population in the United States (adults ages 50-75 years).  The prevalence of lesions in an average-risk population was obtained through a previously published meta-analysis [@heitman2009prevalence]. Based on sensitivities of FIT and MMT in our dataset, we estimate that MMT would detect approximately 40 thousand additional cancers, 1.3 million additional advanced adenomas, and 5.1 million additional non-advanced adenomas compared to using FIT (`r pred_values`). Thus the improved sensitivity of the MMT would increase the total number of true positives identified in the recommended screening population of the United States by approximately 6.5 million. However, due to the lower specificity of MMT, it would also result in an estimated 4.3 million additional false positives compared to FIT. Further studies would be needed to determine whether detection of 6.5 million additional lesions (mostly non-advanced adenomas) would outweigh the added cost of 4.3 million additional false positives.


_**Effect of patient characteristics on model performance.**_ 
Previous studies have identified differences in diagnostic test performance for certain demographic groups or for people taking certain medications [@symonds2015factors;@kapidzic2015gender;@levi2009sensitivity]. Therefore we tested whether the MMT performance differed between patient populations. We found no difference in model performance according to age, BMI, NSAID usage, diabetes, smoking, or previous history of polyps (all p>0.05). However the model was significantly better at differentiating normal from lesion for females than for males (`r pval(gender_pval)`; `r gender`). For females the model detected `r 100*signif(f_sens, 3)`% of lesions with a specificity of `r 100*signif(f_spec, 3)`%. For males the model detected `r 100*signif(m_sens, 3)`% of lesions with a much lower specificity of `r 100*signif(m_spec, 3)`%. The MMT detected `r 100*signif(f_ade_sens, 3)`% of adenomas in females and `r 100*signif(m_ade_sens, 3)`% in males. Despite performing more poorly overall for males, the MMT did have a higher sensitivity for cancer among males (`r 100*signif(m_canc_sens, 3)`%) than females (`r 100*signif(f_canc_sens, 3)`%). The difference in performance between males and females appeared to be due to differences in FIT results rather than differences in the microbiome. After correcting for diagnosis, there was a significant effect of sex on FIT result (`r pval(fit_aov_p)`, two-way ANOVA), but not on the overall structure of the microbiome (PERMANOVA: `r pval(mod_adonis_p)`). 

We have previously shown that incorporating patient metadata into microbiome-based diagnostic models can improve screening accuracy [@zackular2014human]. To test whether the same was true for the MMT we generated a random forest model that combined patients' age, BMI, sex, and smoking status with the OTUs and FIT result from the MMT. The AUC of the ROC curve for this model (`r sprintf('%.3f',demo_roc$auc)`) was not significantly higher than that of the MMT (AUC: `r sprintf('%.3f',les_roc$auc)`, p=`r sprintf('%.2g',roc.test(demo_roc, les_roc)$p.value)`, `r demo_fig`). When the model with patient metadata was set to the same specificity as the MMT (`r round(100*mmt_spec, 1)`%), it did not improve the sensitivity for lesions (`r sprintf('%.1f',100*demo_sens)`%) compared to MMT (`r sprintf('%.1f',100*les_sens_ci[2])`%, p=`r sprintf('%.1g',demo_mc$p.value)`). Thus, contrary to our previous findings, incorporation of patient metadata did not significantly improve the MMT.

###Discussion  

We confirmed previous findings that the gut microbiota can be used to differentiate healthy individuals from those with colonic lesions. Although FIT was better at detecting lesions than a model using only the microbiota, microbiota-based models detected a subset of lesions that were not detected by FIT. This suggested that the two methods could complement each other. Based on this observation we developed a cross-validated random forest model that combined both FIT and the microbiota to detect colonic lesions. The resulting MMT had higher sensitivity than FIT for detecting lesions, especially adenomas. The MMT was also able to detect the majority of cancers missed by FIT. However, the increased sensitivity of MMT was accompanied by a decrease in specificity compared to FIT. With a false positive rate more than three times higher than FIT (`r round(100*(1-mmt_spec), 1)`% versus `r round(100*(1-fit_spec_ci[2]), 1)`%), an annual MMT would result in more colonoscopies than using FIT as the primary screening test. However, the higher sensitivity of the MMT might make it possible to reduce the frequency of screening, thereby offsetting the difference in the number of colonoscopies. Additional studies would be needed identify the appropriate screening inverval and to determine whether the increased number of true positives identified by MMT justify the increased number of false positives. 

It was recently shown that when FIT was combined with host-associated DNA biomarkers, the ability to detect adenomas and carcinomas was significantly improved over FIT alone [@imperiale2014multitarget]. The sensitivity of the host-associated DNA screen was 92.3% for cancer and 42.4% for adenomas with a specificity of 89.8%, all very similar to what we observed with our MMT. Such results support the assertion that because of the large interpersonal variation in markers for adenomas and carcinomas, it is necessary to employ a panel of biomarkers and to use a model that integrates the biomarkers. The accuracy of our model may be further improved by incorporating additional indicators such as the host-associated biomarkers or those targeting specific genes involved in the underlying mechanism of tumorigenesis such as bacterial toxins [@wu2009human;@arthur2012intestinal;@zeller2014potential]. More generally, predictive and diagnostic models for other diseases with a microbial etiology may benefit from a similar approach. For example, we recently demonstrated the ability to detect _Clostridium difficile_ infection based on the composition of the microbiota [@schubert2015antibiotic]. Such models are likely to be useful as microbiota sequencing gains traction as a tool for characterizing health.

Surprisingly most of the OTUs that work well for identifying cancers, including _Fusobacterium nucleatum_ (OTU264), _Peptostroptococcus stomatis_ (OTU310), and _Parvimonas micra_ (OTU281), were excluded from the MMT. This is likely due to these OTUs being positively correlated with FIT (all p<0.001, Spearman correlation), meaning they add little information when used in combination with FIT. Instead the MMT is enriched for OTUs that help detect adenomas. Thus the MMT model relies primarily on FIT for detecting cancer, and uses the microbiota to help identify adenomas undetectable by FIT alone. It is also interesting that most of the OTUs used in the MMT were enriched in normal individuals, suggesting that a loss of beneficial organisms in addition to the emergence of pathogens may be important for colorectal cancer development. Many of the OTUs that were depleted in patients with lesions belonged to the Ruminococcoaceae and Lachnospiraceae families, which contain the predominant producers of butyrate, a short-chain fatty acid with anti-inflammatory and anti-tumorigenic properties [@pryde2002microbiology;@segain2000butyrate;@d1996butyrate;@hague1995apoptosis]. Likewise Zeller et al. observed a depletion of a potential butyrate-producing _Eubacterium spp._ in patients with CRC [@zeller2014potential]. Loss of butyrate or other anti-inflammatory microbial metabolites may contribute to CRC development. These possibilities highlight the need for longitudinal studies to better understand how changes to an individual's microbiome or the metabolic profile of the gut might predispose them to CRC.

Like other groups, we noticed that the microbiota of CRC patients contained higher levels of bacterial taxa traditionally thought of as oral pathogens, including _Fusobacterium_, _Porphyromonas_, _Peptostreptococus_, _Gemella_, _Parvimonas_, and _Prevotella_. Periodontal pathogens have been shown to promote the progression of oral cancer [@gallimidi2015periodontal].  Therefore it is possible that these taxa could influence the progression of CRC by a similar mechanism. These observation may warrant further investigation into a potential link between periodontal diseases and CRC. Furthermore, since the structure of an individual's oral microbiome is correlated with that of the gut [@ding2014dynamics], alterations in the oral community could potentially be a proxy for ongoing or future changes to the gut community.

Although it is exciting that the addition of the microbiota can improve the sensitivity of FIT, further validation is needed prior to clinical adoption. This represents the largest cohort to date, but still only consists of 490 patients. In contrast, the cohort used to validate the Multitarget stool DNA test included 9,989 subjects. Development of a larger cohort will allow us to apply the MMT to a separate validation set. It is also unclear how sensitive the MMT is to variation in sample preparation and processing. Many of the samples included in the current study were collected 1-2 weeks after the subjects’ colonoscopy. A previous study showed that the microbiome quickly returns to normal following colonoscopy [20]. Likewise we found no difference in the microbiome between samples collected prior to or after colonoscopy (PERMANOVA: p=0.45). Regardless, we would have greater confidence in the predictive potential of the microbiota if all samples were collected prior to colonoscopy. Despite these shortcomings, the ability to improve the sensitivity of detecting adenomas suggests that further methods development and validation are warranted.

 
###Conclusions  
Our findings demonstrate the potential for combining the analysis of a patient's microbiota with conventional stool-based tests to improve CRC detection. Using the random forest algorithm it was possible to interpret FIT results in the context of the microbiota. The MMT had higher sensitivity for lesions, regardless of the stages of tumorigenesis. Moreover the model detected the majority of cancers that FIT was unable to detect. The shortcoming of the MMT is its lower specificity. However, the potential value of the MMT is its higher sensitivity, which is the purpose of preventive screening – finding lesions earlier so that cancer would be avoided.
  

**Abbreviations:**  
FIT: fecal immunochemical test  
gFOBT: guaic fecal occult blood test  
OTU: operational taxonomic unit  
AUC: area under the curve  
ROC curve: reciever operating characteristic curve  
MMT: multitarget microbiota test  


**Availability of data and materials:** Raw fastq files and a MIMARKS file are available through the NCBI Sequence Read Archive [SRP062005]. The exact data processing steps for going from the raw sequence data to the final manuscript is available at http://www.github.com/SchlossLab/Baxter_glne007Modeling_2015.

**Competing interests:** The authors declare no competing financial interests.  

**Author Contributions:** All authors were involved in the conception and design of the study. NTB processed samples and analyzed the data.  All authors interpreted the data. NTB and PDS wrote the manuscript.  All authors reviewed and revised the manuscript.

**Funding:** This study was supported by funding from the National Institutes of Health to P. Schloss (R01GM099514, P30DK034933) and to the Early Detection Research Network (U01CA86400).

**Acknowledgements:** The authors thank the Great Lakes-New England Early Detection Research Network for providing the fecal samples that were used in this study. 


###Figures   
**Figure 1. Microbiota-based models can complement FIT.** (A,C) ROC curves for distinguishing healthy patients from those with adenoma (A) or cancer (C) based on FIT or a microbiota-based random forest model. Open circles show the sensitivity and specifity of FIT with a 100 ng/ml cutoff.  Black points show the sensitivity and specificity of the microbiota-based models at the chosen cutoffs.  (B,D) Results of FIT and a microbiota-based model for each adenoma (B) or cancer (D) sample. Dotted lines represent the cutoffs for each test. Points are shaded based on whether the lesion was detected by both tests (black), one of the two tests (grey), or neither test (white).
  
**Figure 2. Comparing MMT to FIT.** (A) ROC Curves for the MMT (solid lines) or FIT (dashed lines) for distinguishing normal from any lesion (dark red), normal from cancer (red) and normal from adenoma (orange). Filled dots show the sensitivity and specificity of the MMT at the optimal cutoff (0.622).  Open dots show the sensitivity and specificity of FIT at the 100 ng/ml cutoff. (B,C) Stripcharts showing the results for FIT (B) and the MMT (C). Dashed lines show the cutoff for each test. Points with a FIT result of 0 are jittered to improve visibility.
 
**Figure 3. Relationship between FIT and MMT for each sample.** (A) Scatterplot of MMT and FIT results for each sample.  Dashed lines show the cutoff for each test. Points with a FIT result of 0 are jittered to improve visibility. (B) Stripchart of MMT results for samples separated by binary FIT result.

**Figure 4. Sensitivities for FIT and MMT for each stage of tumor development with matching specificities.**  The cutoff for FIT was reduced to 7 ng/ml to match the specificity of the MMT. Sensitivities were compared using the method proposed by Pepe et al. (* = p<0.05, 1000 bootstrap replicates).
 
###Tables  
**Table 1. Sensitivities and specificities for FIT and MMT.** The 95% confidence intervals were computed with 2000 stratified bootstrap replicates.

**Table 2. Estimated number of true positives detected in average risk population.** Number of true positives identified through FIT and MMT in the United States in adults 50-75 years of age, based on published estimates of CRC prevalence. The sensitivities for FIT (100 ng/ml cutoff) on advanced and non-advanced adenomas were `r round(100*advAde_fit_sens, 1)`% and `r round(100*nonAdv_fit_sens, 1)`%, respectively.


###Additional Files  
**Additional file 1: Figure S1. Random forest feature selection for detecting adenomas.**  (A) Change in AUC with varying number of variables in the random forest model.  The model with the highest AUC contained 22 OTUs. (B) Importance of each OTU in the model as measured by mean decrease accuracy when the OTU is removed from the model. (C) Relative abundance of the most discriminatory OTUs in adenoma and normal samples.

**Additional file 2: Figure S2. Random forest feature selection for detecting cancers.**  (A) Change in AUC with varying number of variables in the random forest model.  The model with the highest AUC contained 34 OTUs. (B) Importance of each OTU in the model as measured by mean decrease accuracy when the OTU is removed from the model. (C) Relative abundance of the most discriminatory OTUs in cancer and normal samples.
 
**Additional file 3: Figure S3. Bacterial OTUs in MMT.** (left) Importance of each OTU used in the MMT as measured by the mean decrease in the Gini index when the OTU is removed from the model. (right) Stripchart of the relative abundances of each OTU in the MMT with black lines at the medians.
  
**Additional file 4: Figure S4. MMT performance by sex.** ROC curves (left) and stripchart (right) of MMT results separated by sex. 

**Additional file 5: Figure S5. MMT with patient metadata.** ROC curves for distinguishing normal from lesion using FIT, the MMT, or the MMT with metadata.

###References
